catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -263,6 +235,7 @@ public void updateCallback(Map<String, Object> properties) {
         setAttributeOverrides(Arrays.asList(parameterArray));
       }
 
+      destroy(0);
```

Comment:
❗️ Because `destroy()` is now async, there is a high chance that you will start initializing before it has completed its removal. You probably need to get a hold of the `CompletableFuture` from the `runAsync()` in destroy and wait for it before initiating `init()`